packages/story-editor/src/components/library/panes/media/common/innerElement.js

```diff
@@ -109,11 +110,35 @@ function InnerElement({
   isMuted,
 }) {
   const newVideoPosterRef = useRef(null);
+  // Track if we have already set the dragging resource.
+  const hasSetResourceTracker = useRef(null);
+
+  // Note: This `useDropTargets` is purposefully separated for the one below since it
+  // uses a custom function for checking for equality and is meant for `handleDrag` and `handleDrop` only.
+  const { handleDrag, handleDrop } = useDropTargets(
+    ({ actions: { handleDrag, handleDrop } }) => ({
+      handleDrag,
+      handleDrop,
+    }),
+    () => {
+      // If we're dragging this element, always update the actions.
+      if (hasSetResourceTracker.current) {
+        return false;
+        // If we're rendering the first time, init `handleDrag` and `handleDrop`.
+      } else if (hasSetResourceTracker.current === null) {
+        hasSetResourceTracker.current = false;
+        return false;
+      }
+      // Otherwise ignore the changes in the actions.
+      return true;
+    }
```

Comment:
Uh, that's mighty fancy. We should probably use similar tricks elsewhere. I hadn't even considered using the equality function for this purpose... could definitely save some re-renders!